### PR TITLE
Benchmarks - Add LLaMA-2 Models

### PR DESCRIPTION
**Pull request by dpower4**
Added llama benchmark - training and inference in accordance with the existing pytorch models implementation like gpt2, lstm etc.

- added llama fp8 unit test for better code coverage, to reduce memory required
- updated transformers version >= 4.28.0 for LLamaConfig
- set tokenizers version <= 0.20.3 to avoid 0.20.4 version [issues](https://github.com/huggingface/tokenizers/issues/1691) with py3.8
- added llama2 to tensorrt
- llama2 tests not added to test_tensorrt_inference_performance.py due to large memory requirement for worker gpu. tests validated separately on gh200

### CODE REVIEW

File: superbench/benchmarks/micro_benchmarks/_export_torch_to_onnx.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def export_torchvision_model(self, model_name, batch_size=1):
             model,
             dummy_input,
             file_name,
-            opset_version=10,
+            opset_version=14,
```
Comment (abuccts) [inline superbench/benchmarks/micro_benchmarks/_export_torch_to_onnx.py:141]: are changes in this file needed for this PR on llama2?

**Review by abuccts — COMMENTED** [review]
pls use `python3 setup.py lint` to check the format and run `python3 setup.py format` to format and code

File: superbench/benchmarks/micro_benchmarks/_export_torch_to_onnx.py (hunk 1 above)
Comment (dpower4) [reply superbench/benchmarks/micro_benchmarks/_export_torch_to_onnx.py:141]: with the updated transformer package(>=4.28.0), the pytorch models use the scaled_dot_product_attention

else the error in unit testing test_tensorrt_inference_performance.py:

torch.onnx.errors.UnsupportedOperatorError: Exporting the operator 'aten::scaled_dot_product_attention' to ONNX opset version 10 is not supported. Support for this operator was added in version 14, try exporting with this version.

**Comment by dpower4** [comment]
@abuccts can I get access to the unit test logs.

**Review by guoshzhao — APPROVED** [review]

**Comment by dpower4** [comment]
pytorch-llama validation on gh200.
Screenshot 2024-11-21 at 6 55 33 PM
Screenshot 2024-11-21 at 6 53 56 PM

**Comment by dpower4** [comment]
tokenizers cargo issue: https://github.com/huggingface/tokenizers/issues/1691

File: setup.py [hunk 2]
```diff
@@ -211,7 +211,8 @@ def run(self):
             'torch': [
                 'torch>=1.7.0a0',
                 'torchvision>=0.8.0a0',
-                'transformers>=4.3.3, <4.23.0',
+                'tokenizers <= 0.20.3',
```
Comment (abuccts) [inline setup.py:214]: move it before torch? the list is sorted alphabetically, and remove space around "<=" to keep consistent?

File: superbench/benchmarks/base.py [hunk 3]
```diff
@@ -89,7 +89,11 @@ def get_configurable_settings(self):
         Return:
             All configurable settings in raw string.
         """
-        return self._parser.format_help().strip()
+        message = self._parser.format_help().strip()
+        # Ensure consistent header across diff python argparse format_help output.
+        if 'options:' in message:
+            message = message.replace('options:', 'optional arguments:')
+        return message
```
Comment (abuccts) [inline superbench/benchmarks/base.py:96]: pls revert this change, it's fixed in #671 by setting `parser._optionals.title`

**Review by abuccts — APPROVED** [review]
